phi/playground/router.py

```diff
@@ -94,47 +100,103 @@ def chat_response_streamer(
             run_response_chunk = cast(RunResponse, run_response_chunk)
             yield run_response_chunk.model_dump_json()
 
-    def process_image(file: UploadFile) -> List[Union[str, Dict]]:
-        content = file.file.read()
-        encoded = base64.b64encode(content).decode("utf-8")
+    def process_image(files: List[UploadFile]) -> List[List[Union[str, Dict]]]:
+        images = []
+        for file in files:
+            content = file.file.read()
+            encoded = base64.b64encode(content).decode("utf-8")
 
-        image_info = {"filename": file.filename, "content_type": file.content_type, "size": len(content)}
+            image_info = {"filename": file.filename, "content_type": file.content_type, "size": len(content)}
+            images.append([encoded, image_info])
 
-        return [encoded, image_info]
+        return images
 
     @playground_router.post("/agent/run")
-    def agent_run(body: AgentRunRequest):
-        logger.debug(f"AgentRunRequest: {body}")
-        agent = get_agent_by_id(body.agent_id, agents)
+    def agent_run(
+        message: str = Form(...),
+        agent_id: str = Form(...),
+        stream: bool = Form(True),
+        monitor: bool = Form(False),
+        session_id: Optional[str] = Form(None),
+        user_id: Optional[str] = Form(None),
+        files: Optional[List[UploadFile]] = File(None),
+    ):
+        logger.debug(f"AgentRunRequest: {message} {agent_id} {stream} {monitor} {session_id} {user_id} {files}")
+        agent = get_agent_by_id(agent_id, agents)
         if agent is None:
             raise HTTPException(status_code=404, detail="Agent not found")
 
-        if body.session_id is not None:
-            logger.debug(f"Continuing session: {body.session_id}")
+        if files:
+            if agent.knowledge is None:
```

Comment:
Why is this required? The upload should just be about adding it to the message sent to the agent

Reply:
Adding it to the knowledgebase could be a really nice side benefit, but it should be optional I think

Reply:
I don't think we can pass on files to messages right now